uaclient/config.py

```diff
@@ -160,14 +161,18 @@ def data_path(self, key=None, private=True):
             return os.path.join(data_dir, self.data_paths[key])
         return os.path.join(data_dir, key)
 
-    def delete_cache(self):
-        """Remove all configuration cached response files class attributes."""
+    def delete_cache(self, key=None):
```

Comment:
I think a clearer API would be to have separate methods for "delete everything" and "delete this one thing"; very easy to accidentally delete everything by passing something False-ish as `key` to this current one.

(This doesn't need to happen in this PR; once it lands let's open up a techdebt issue.)

Reply:
separate config.delete_cache_key method is now published.